File: CSTH/OLD/test2/pack1/Sensor.java
package pack1;
import java.util.*;
import java.text.*;

public class Sensor {
  private static DecimalFormat df;
  public ToaDo center;
  public double radius;
  private List<Integer> targetCovered;
  private int INVALID = -1;

  public Sensor(ToaDo tam,double rad){
    setCenter(tam);
    radius = rad;
  }
  public void setCenter(ToaDo tam){
    center = new ToaDo(tam);
    targetCovered = new ArrayList<Integer>();
  }
  public String toCircle(){
    df = new DecimalFormat("#.###");
    return "(x-" + df.format(center.getX()) +")^2 +(y-" + df.format(center.getY()) + ")^2 = " + df.format(radius * radius / 4) ;
  }
  public int getTargetCount(){
    return targetCovered.size();
  }
  public int isHave(int target){
  /*
    Tra ve INVALID neu khong co
    Tra ve index neu co
  */
    for (int i=0;i < targetCovered.size();i++)
      if (targetCovered.get(i) == target) return i;
    return INVALID;
  }
  public void add(int target){
    int id = isHave(target);
    if (id == INVALID) 
      targetCovered.add(target);
  }
  public void remove(int target){
    int id = isHave(target);
    if (id != INVALID) targetCovered.remove(id);
  }
  public boolean isCover(ToaDo target){
    double EPS = (double) 1e-10;
    if (center.khoangCach(target) - radius < EPS) return true;
    return false;
  }
  public Integer[] returnList(){
    Integer[] list = new Integer[targetCovered.size()];
    for (int i=0;i < targetCovered.size();i++)
      list[i] = targetCovered.get(i);
    return list;
  }
}

File: CSTH/NEW/test2/pack1/Sensor.py
from pack1.ToaDo import ToaDo

class Sensor:
	def __init__(self, tam: ToaDo, rad) -> None:
		self.center: ToaDo = None
		self.targetCovered = []

		self.setCenter(tam)
		self.radius = rad
	
	def setCenter(self, tam: ToaDo):
		self.center = ToaDo(tam)
		self.targetCovered = []
	
	def toCircle(self):
		return f'(x-{self.center.getX():.3f})^2 +(y-{self.center.getY():.3f})^2 = {self.radius * self.radius/4:.3f}'

	def getTargetCount(self):
		return len(self.targetCovered)


	def add(self, target):
		if target not in self.targetCovered:
			self.targetCovered.append(target)

	def remove(self, target):
		if target in self.targetCovered:
			self.targetCovered.remove(target)
	
	def isCover(self, target: ToaDo):
		EPS = 1e-10

		return self.center.khoangCach(target) - self.radius < EPS

	def returnList(self):
		lst = self.targetCovered.copy()

		return lst

